python/grpc_opentracing/__init__.py

```diff
@@ -0,0 +1,106 @@
+import abc
+import enum
+
+import six
+
+import grpc
+
+
+class ActiveSpanSource(six.with_metaclass(abc.ABCMeta)):
+    """Provides a way to access an the active span."""
+
+    @abc.abstractmethod
+    def get_active_span(self):
+        """Identifies the active span.
+
+    Returns:
+      An object that implements the opentracing.Span interface.
+    """
+
+
+@enum.unique
+class ClientRequestAttribute(enum.Enum):
+    """Optional tracing tags available on the invocation-side.
+
+  Attributes:
+    HEADERS: The initial :term:`metadata`.
+    METHOD_TYPE: Type of the method invoked.
+    METHOD_NAME: Name of the method invoked.
+    DEADLINE: The length of time in seconds to wait for the computation to
+      terminate or be cancelled.
+  """
+    HEADERS = 0
+    METHOD_TYPE = 1
+    METHOD_NAME = 2
+    DEADLINE = 3
+
+
+@enum.unique
+class ServerRequestAttribute(enum.Enum):
+    """Optional tracing tags available on the service-side.
+
+  Attributes:
+    HEADERS: The initial :term:`metadata`.
+    METHOD_TYPE: Type of the method invoked.
+    METHOD_NAME: Name of the method invoked.
+    DEADLINE: The length of time in seconds to wait for the computation to
+      terminate or be cancelled.
+  """
+    HEADERS = 0
+    METHOD_TYPE = 1
+    METHOD_NAME = 2
+    DEADLINE = 3
+
+
+def open_tracing_client_interceptor(tracer,
+                                    active_span_source=None,
+                                    log_payloads=False,
+                                    traced_attributes=None):
```

Comment:
thinking more deeply about this, IMO it might be better to express this functionality as a SpanDecorator function that gives the caller the flexibility to modify the RPC Span in any way (and provides RPC context information like headers, method type/name, deadline, etc in the decorator function arguments somehow).